### DISCUSSION TITLE
Renovate GitHub App stopped working (repository not found)

### DISCUSSION THREAD

**Discussion by nicholasdille** [renovatebot/renovate#21847] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Renovate should continue to operate on the repository.

Repository: https://github.com/nicholasdille/docker-setup

### What you tried so far.

When I ask for another run of renovate on the repository from the dependency dashboard, it works for a few runs and then stops again with the same error.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
I've added https://github.com/renovatebot/renovate/commit/de3cb95e91569768608df6a10d7e166f221c1243 so we can see if there's any hints there as to why no repository is being returned.

**Reply by rarkins** [reply]
Naturally, the next time Renovate ran on that repo it succeeded

**Reply by rarkins** [reply]
```json
{
  "level": 20,
  "res": {
    "errors": [
      {
        "type": "RATE_LIMITED",
        "message": "API rate limit exceeded for installation ID 9306189."
      }
    ]
  },
  "msg": "No repository returned",
  "time": "2023-04-27T19:58:46.771Z"
}
```

**Reply by rarkins** [reply, marked as answer]
Created #21861

**Reply by nicholasdille** [reply]
Thanks for the quick response!

Do you know why the renovate app did not recover? It only started working after I did something manually. It feels like it got stuck after the error.

Is the API rate limited counted per repository? Am I causing this somehow?

**Reply by rarkins** [reply]
A not found repo gets marked as disabled in the backend

**Reply by nicholasdille** [reply]
That makes sense. Are you planning to change the behaviour once the rate limiting error is handled correctly?

**Reply by rarkins** [reply]
Yes, as soon as that bug issue I linked to is fixed, we should exit with a "temporary" error (rate limiting) instead of "permanent" (not found), which means it will be retried the next time like normal instead of taken out of service

**Reply by nicholasdille** [reply]
I have been thinking... Did I cause the API rate limit issue because I have so many dependencies? Should I introduce rate limiting in renovate?

**Reply by rarkins** [reply]
Maybe, but it's a bit strange to see it happen during the repo init instead of later when dependencies are being looked up

**Reply by nicholasdille** [reply]
In the last few days I have played with schedules and PR rate limiting. Have you changed anything in the Renovate app or did my changes improve the situation?

**Reply by rarkins** [reply]
I haven't changed anything

**Reply by nicholasdille** [reply]
I noticed that Renovate throws a warning related to a rate limit and continues to operate on the repo based on the configured schedule. Look like this has been fixed 🚀

**Comment by nicholasdille** [comment]
BTW, I love what you are doing with renovate. I am speaking in favour of it any time I can :-D

**Reply by rarkins** [reply]
Thanks! Word of mouth and engaged users is the best way we've grown the project
